posthog/api/team.py

```diff
@@ -224,12 +224,21 @@ def team(self):
 
     def perform_destroy(self, team: Team):
         team_id = team.pk
-        delete_bulky_postgres_data(team_ids=[team_id])
-        AsyncDeletion.objects.create(
-            deletion_type=DeletionType.Team, team_id=team_id, key=str(team_id), created_by=cast(User, self.request.user)
-        )
-        with mute_selected_signals():
-            super().perform_destroy(team)
+        with transaction.atomic():
```

Comment:
I'm not sure we want this.

In theory yes - doing it in one transaction is ideal. But what if `super().perform_destroy(team)` below takes 100s? Could this take down ingestion if any accidental locking starts blocking some writes to person table or cause other issues.

Reply:
Fair point. Alternative proposal: I put `AsyncDeletion` creation at the end of each method, so that we only queue deletion of associated data when we're sure the parent is gone. How does that look?